File: Creational/Builder/Python/car_builder.py
# car_builder.py

from car import Car

class CarBuilder:
  def __init__(self):
    self.engine = "Unknown Engine"
    self.has_gps = False
    self.has_sunroof = False
    self.seats = "Standard"

  def set_engine(self, engine):
    self.engine = engine
    return self

  def set_gps(self, has_gps):
    self.has_gps = has_gps
    return self

  def set_sunroof(self, has_sunroof):
    self.has_sunroof = has_sunroof
    return self

  def set_seats(self, seats):
    self.seats = seats
    return self

  def build(self):
    return Car(self.engine, self.has_gps, self.has_sunroof, self.seats)

File: Creational/Builder/JavaScript/carBuilder.js
// carBuilder.js

const Car = require("./car");

class CarBuilder {
  constructor() {
    this.engine = "Unknown Engine";
    this.hasGPS = false;
    this.hasSunroof = false;
    this.seats = "Standard";
  }

  setEngine(engine) {
    this.engine = engine;
    return this;
  }

  setGPS(hasGPS) {
    this.hasGPS = hasGPS;
    return this;
  }

  setSunroof(hasSunroof) {
    this.hasSunroof = hasSunroof;
    return this;
  }

  setSeats(seats) {
    this.seats = seats;
    return this;
  }

  build() {
    return new Car(this.engine, this.hasGPS, this.hasSunroof, this.seats);
  }
}

module.exports = CarBuilder;
